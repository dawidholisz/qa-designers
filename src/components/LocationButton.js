import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

const LocationButton = props => {
  return (
    <FontAwesomeIcon
      className={classNames('set-location-button', { '-night': !props.isDay })}
      icon="search-location"
      onClick={() => {
        props.onClick()
      }}
      size="lg"
    />
  )
}

LocationButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  isDay: PropTypes.bool,
}

export default LocationButton
