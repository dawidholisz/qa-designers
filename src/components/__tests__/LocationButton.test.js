import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import LocationButton from '../LocationButton'

const renderComponent = ({ onClick, isDay }) => {
  return render(<LocationButton isDay={isDay} onClick={onClick} />)
}
const props = {
  onClick: jest.fn(),
  isDay: true,
}
jest.mock('@fortawesome/react-fontawesome', () => ({
  FontAwesomeIcon: ({ className, onClick }) => {
    return (
      <p
        className={className}
        onClick={() => {
          onClick()
        }}
      >
        Icon
      </p>
    )
  },
}))

describe('<LocationButton/>', () => {
  it('render LocationButton', () => {
    const { container } = renderComponent(props)
    expect(container.firstChild).toMatchSnapshot()
  })
  it('render night LocationButton', () => {
    const { container } = renderComponent({ ...props, isDay: false })
    expect(container.firstChild).toMatchSnapshot()
  })
  // it('click on LocationButton', () => {
  //   const { getByText, debug } = renderComponent({ props })
  //   debug()
  //   fireEvent.click(getByText('Icon'))
  //   expect(props.onClick).toHaveBeenCalled()
  // })
})
