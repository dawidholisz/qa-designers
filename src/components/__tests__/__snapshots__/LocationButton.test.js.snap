// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationButton/> render LocationButton 1`] = `
<p
  class="set-location-button"
>
  Icon
</p>
`;

exports[`<LocationButton/> render night LocationButton 1`] = `
<p
  class="set-location-button -night"
>
  Icon
</p>
`;
